src/Data/LLVM/BitCode/Parse.hs

```diff
@@ -145,13 +151,13 @@ setRelIds b = Parse $ do
   put $! ps { psValueTable = (psValueTable ps) { valueRelIds = b }}
 
 getRelIds :: Parse Bool
-getRelIds  = Parse $ do
```

Comment:
Why did all these change? Can we split that into a different commit?

Reply:
The `getLastLoc` change is actually related to MonadFail.  I needed the `fail` to occur in the `Parse` monad rather than the underlying monad, which doesn't have a `fail` instance.  I think it was always a bug to fail that way.

I did this change in `getRelIds` (and several other places) for uniformity, but maybe it would be better to leave it alone.